File: programming/fibonacci-sequence-via-for-loop/fibModified.py
import time

def fib():

    def getFib(n_int):
        if n_int > 1:
            fib = getFib(n_int-1) + getFib(n_int-2);
        elif n_int == 1:
            fib = 1
        else:
            fib = 0
        return fib

    n = input('Please enter a non-negative integer or type stop: ')
    if n=="stop":
        return None
    else:
        n = int(n);
        tstart = time.process_time()
        print( "fib({}) = {}".format(n,getFib(n)) );
        tend = time.process_time()
        print( "average runtime: {} seconds".format(tend-tstart) );
        fib()
        return None

File: programming/fibonacci-sequence-via-for-loop/fibModified.m
function fibModified()

    n = input('Please enter a non-negative integer or type stop: ','s');
    if strcmp(n,'stop')
        return
    else
        n = str2double(n);
        if isreal(n)
            if n>=0 && round(n)==n
                disp([char(9), 'fib(',num2str(n),') = ',num2str(getFib(n))]);
                runtime = timeit( @()getFib(n) );
                disp([char(9), 'average runtime: ',num2str(runtime), ' seconds']);
                fib()
                return
            end
        end
        disp('The input argument is not a non-negative integer!');
        fib()
    end
    
    function fib = getFib(n_int)
        if n_int == 0
            fib = 0;
        elseif n_int == 1
            fib = 1;
        else
            fib = getFib(n_int-1) + getFib(n_int-2);
        end
    end

end